dbms/src/Formats/ODBCDriver2BlockOutputStream.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Core/Block.h>
+#include <Formats/FormatFactory.h>
+#include <Formats/ODBCDriver2BlockOutputStream.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteHelpers.h>
+
+
+#include <Core/iostream_debug_helpers.h>
+
+
+namespace DB
+{
+ODBCDriver2BlockOutputStream::ODBCDriver2BlockOutputStream(
+    WriteBuffer & out_, const Block & header_, const FormatSettings & format_settings)
+    : out(out_), header(header_), format_settings(format_settings)
+{
+}
+
+void ODBCDriver2BlockOutputStream::flush()
+{
+    out.next();
+}
+
+void writeODBCString(WriteBuffer & out, const std::string & str)
+{
+    writeIntBinary(Int32(str.size()), out);
+    out.write(str.data(), str.size());
+}
+
+void writeODBCString(WriteBuffer & out, const char * cstr)
+{
+    std::string str{cstr};
+    writeODBCString(out, str);
+}
+
+void ODBCDriver2BlockOutputStream::write(const Block & block)
+{
+    const size_t rows = block.rows();
+    const size_t columns = block.columns();
+    String text_value;
+
+    for (size_t i = 0; i < rows; ++i)
+    {
+        for (size_t j = 0; j < columns; ++j)
+        {
+            text_value.resize(0);
+            const ColumnWithTypeAndName & col = block.getByPosition(j);
+
+            if (col.column->isNullAt(i))
+            {
+                writeIntBinary(Int32(-1), out);
+            }
+            else
+            {
+                {
+                    WriteBufferFromString text_out(text_value);
+                    col.type->serializeText(*col.column, i, text_out, format_settings);
+                }
+                writeODBCString(out, text_value);
+            }
+        }
+    }
+}
+
+void ODBCDriver2BlockOutputStream::writePrefix()
```

Comment:
I cannot understand the motivation behind this format.

Reply:
expandable header without breaking format - we can send any column flags, for example length for display_size or something else